metabase/metabase
# Unify semicolon and comment handling in nested native queries

Closes #29168

### Description

Prior to this PR comments were removed only from _native source queries of mbql queries_ in `fetch-source-query/source-query`. Semicolons were removed by `sql.qp/make-nestable-sql` and `sql.qp.deprecated/SQLSourceQuery`'s `to-sql` implementation. Semicolon removal was handled also for _native card references in native queries_. 

This PR unifies the logic for comment and semicolon removal in `sql.qp/make-nestable-sql`, which results in comments being also removed from _native card references in native queries_.

Drivers of both honeysql versions were considered and the tests were updated accordingly.

- `card-id-native-source-queries-test` was adjusted to run with any driver with `:sql` parent. This verifies, that _native source queries of mbql queries_ are handled correctly with drivers of both honeysql versions.
- Testcase from `card-id->source-query-and-metadata-test` was merged into the `substitute-referenced-card-query-test`. This covers comment and semicolon removal from _native card references in native queries_.

### How to verify

See #29168 for reproduction or run at least following tests with honeysql 1 and 2 driver:
- `card-id-native-source-queries-test`,
- `substitute-referenced-card-query-test`.

Change was tested locally with: `DRIVERS=h2,sqlserver,presto-jdbc clj -X:dev:ee:ee-dev:drivers:drivers-dev:test`.

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30677)
